x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts

```diff
@@ -83,22 +83,20 @@ export function initSpacesOnPostAuthRequestInterceptor({
 
         const statusCode = wrappedError.statusCode;
 
-        // If user is not authorized, or the space cannot be found, allow them to select another space
-        // by redirecting to the space selector.
-        const shouldRedirectToSpaceSelector = statusCode === 403 || statusCode === 404;
-
-        if (shouldRedirectToSpaceSelector) {
-          log.debug(
-            `Unable to navigate to space "${spaceId}", redirecting to Space Selector. ${error}`
-          );
-          return response.redirected({
-            headers: {
-              location: getSpaceSelectorUrl(serverBasePath),
-            },
-          });
-        } else {
-          log.error(`Unable to navigate to space "${spaceId}". ${error}`);
-          return response.customError(wrappedError);
+        switch (statusCode) {
+          case 403:
+            log.debug(`User unauthorized for space "${spaceId}". ${error}`);
+            return response.forbidden();
```

Comment:
@jportner I implemented your suggestion, but changed it to `response.fobidden()` instead of `response. unauthorized()` (i.e. 403 instead of 401) so that the `onPreResponse` hook will pick it up and show the new screen.

I hope you don't mind that I changed it to a switch-statement?

Reply:
Sounds good to me!